image_services/opera/update_opera_uris.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import csv
+import json
+import logging
+import os
+from pathlib import Path
+
+import boto3
+import requests
+
+S3_CLIENT = boto3.client('s3')
+log = logging.getLogger(__name__)
+
+
+def query_cmr(polarization):
+    session = requests.Session()
+    search_url = 'https://cmr.earthdata.nasa.gov/search/granules.umm_json'
+
+    params = {
+        'short_name': 'OPERA_L2_RTC-S1_V1',
+        'attribute[]': f'string,POLARIZATION,{polarization}',
+        'page_size': 2000,
+    }
+    headers = {}
+    vsis3_uris = []
+    while True:
+        response = session.get(search_url, params=params, headers=headers)
+        response.raise_for_status()
+        for granule in response.json()['items']:
+            for url in granule['umm']['RelatedUrls']:
+                if url['URL'].startswith('s3://') and url['URL'].endswith(f'{polarization}.tif'):
+                    vsis3_uris.append(url['URL'].replace('s3://', '/vsis3/'))
+                    break
+        if 'CMR-Search-After' not in response.headers:
+            break
+        headers['CMR-Search-After'] = response.headers['CMR-Search-After']
+        return vsis3_uris
```

Comment:
Is this ```while True:``` loop necessary? The code still seems to work without it.

Reply:
@asjohnston-asf I can't remember what the purpose of that `True` statement is. Was it to handle some sort of CMR outage?

Reply:
@AndrewPlayer3 The `while True:` is how we handle multiple pages. I had to double-check with Andrew Johnston about this. MM link with his explanation is here: https://chat.asf.alaska.edu/asf/pl/66kujezwrtg67kd1gcu1mrmtqw